sandbox-service/src/main/java/com/exadel/sandbox/service/impl/TagServiceImpl.java

```diff
@@ -47,6 +53,23 @@
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public void deleteTagById(Long tagId) {
+        Optional<Tag> tagOptional = repository.findById(tagId);
+        if(tagOptional.isPresent()){
+            Set<Event> events = tagOptional.get().getEvents();
+            if(events.isEmpty() || events.size()==0){
+                repository.delete(tagOptional.get());
+            }else{
+                throw new ResponseStatusException(HttpStatus.NOT_ACCEPTABLE,
+                        "You cannot delete tag. Tag is uses");
+            }
+        }else{
+            throw new ResponseStatusException(HttpStatus.NOT_ACCEPTABLE,
+                    "Tag is not exists");
+        }
+    }
+
```

Comment:
Does these changes belong to the scope of your task?
I am asking because they are part of my task  (tag api).
Was this agreed with me?